staging/src/k8s.io/controller-manager/pkg/leadermigration/options/options.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"fmt"
+
+	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/controller-manager/config"
+	"k8s.io/controller-manager/config/v1alpha1"
+	"k8s.io/controller-manager/pkg/leadermigration"
+	migrationconfig "k8s.io/controller-manager/pkg/leadermigration/config"
+)
+
+type LeaderMigrationOptions struct {
+	Enabled                   bool
+	ControllerMigrationConfig string
+}
+
+// DefaultLeaderMigrationOptions returns a LeaderMigrationOptions with default values.
+func DefaultLeaderMigrationOptions() *LeaderMigrationOptions {
+	return &LeaderMigrationOptions{}
+}
+
+// AddFlags adds all flags related to leader migration to given flag set.
+func (o *LeaderMigrationOptions) AddFlags(fs *pflag.FlagSet) {
+	if o == nil {
+		return
+	}
+	fs.BoolVar(&o.Enabled, "enable-leader-migration", false, "Whether to enable controller leader migration.")
+	fs.StringVar(&o.ControllerMigrationConfig, "leader-migration-config", "", "Path to the config file for controller leader migration. Leave empty to use default value.")
+}
+
+// ApplyTo applies the options of leader migration to generic configuration.
+func (o *LeaderMigrationOptions) ApplyTo(cfg *config.GenericControllerManagerConfiguration) error {
+	if o == nil {
+		return nil
+	}
```

Comment:
Are we supporting nil LeaderMigrationOptions?  I was expecting that the default would be required at a minimum. I don't feel strongly either way, but maybe add a comment explaining somewhere?

Reply:
LeaderMigrationOptions will never be `nil`. This is a result of consulting other implementation of `ApplyTo` here https://github.com/kubernetes/kubernetes/blob/c200a8f9b75072482ed4c4164db8d260ae4ab9ac/staging/src/k8s.io/controller-manager/options/generic.go#L49 (this case, `o` can not be `nil` either)
Will remove.